### ISSUE TITLE
Fix the docs about "Unicode characters"

### ISSUE THREAD

**Issue by m13253** [nim-lang/Nim#4795]
## Description

The procedure [`runeLen`](http://nim-lang.org/docs/unicode.html#runeLen,string) is defined as "Returns the number of Unicode characters of the string s".
However "Unicode character" is an ambiguous term.

As is [UTF-8 Everywhere](http://utf8everywhere.org/#characters) written, a "character" could be defined as: a code point, a code unit, an abstract character, an encoded character, a user-perceived character, a grapheme cluster, or a glyph.
Actually `runeLen` counts the number of "code points", while a single code point may not correspond to an glyph on the screen.
This might mislead some developers, when they develop a word processing software or something like that, into assuming pressing "←" once will move the cursor back "1 rune".

In addition, the definition of [`Rune`](http://nim-lang.org/docs/unicode.html#Rune) is also wrong. It says "type that can hold any Unicode character", but it can not even hold a single "ю́ cyrillic small letter yu with acute", or a single "👨‍👩‍👧‍👦 emoji family with a man, a woman, a girl, and a boy".
## Suggestions
1. Change the definition of "Rune" to "type that can hold a single Unicode code point. A Rune may be composed with other Runes to a character on the screen".
2. Change the definition of "Rune16" to "type that can hold a single UTF-16 encoded character. A single Rune16 may not be enough to hold an arbitrary Unicode code point"
3. Search all through the documentation for the word "character". Fix them according to the exact behavior.
## Some personal complainments

I, as a CJK developer, think Unicode support is important, as well as not misleading developers to form wrong concepts.
Microsoft has previously fooled Notepad users to think "every non-Unicode encoding is American standard (ANSI)", and even fooled developers to think "GB18030 is not a Unicode encoding" by providing a function named [`BytesToUnicode`](https://msdn.microsoft.com/en-us/library/windows/desktop/dd317724%28v=vs.85%29.aspx) (later removed).

**TL;DR:** Please be serious about Unicode concepts. I hope Nim developers could be much more Unicode-aware than the average developers.

Thank you.

**Comment by johnnovak** [comment]
I couldn't agree more 💯 
While I agree that "rune" sounds way cooler, "code point" is the proper term to use.
And yes, code point != character.

Whoever will work on the fix, this is also a good (and essential) article on the matter:
http://www.joelonsoftware.com/articles/Unicode.html

Another very important thing to note is that all functions should work correctly with strings containing a mixture of 1, 2, 3 and 4-byte Unicode code points, there should be test cases for that.
